### DISCUSSION TITLE
Os melhores desktops para desenvolvimento

### DISCUSSION THREAD

**Discussion by kvnol** [frontendbr/forum#817] [Dicas]
Não sei vocês, mas eu sempre passei raiva com meu computador e acho que chegou na hora de montar um novo. E queria propor um exercício para não ajudar só a mim e sim a todos que querem montar uma nova máquina para desenvolver.

O exercício seria montarmos, e chegar numa conclusão de, **3 configurações** (básica, média e avançada) para desenvolvimento web e, quem sabe, até prototipação de UI.

**Comment by kvnol** [comment]
Minha primeira sugestão seria para monitor, que foi um grande upgrade para meu setup, o [Monitor LG LED 23´ IPS D-Sub, HDMI, Full HD 23MP55HQ-P](https://www.kabum.com.br/produto/54259/monitor-lg-led-23-ips-d-sub-hdmi-full-hd-23mp55hq-p-preto/?tag=Monitor%20LG%20LED%2023).

Creio que R$ 629,90 (o preço deste LG) seja preço para monitores baixo custo, não? Tendo em vista que temos monitores que ultrapassam a casa dos 2 mil reais.

**Comment by silvaMatheus** [comment]
Recentemente comprei esse ultraWide [Monitor Gamer LG 29 Full HD IPS LED 1ms UltraWide 21:9 HDMI Preto 29UM69G](https://www.kabum.com.br/produto/90224/monitor-gamer-lg-29-full-hd-ips-led-1ms-ultrawide-21-9-hdmi-preto-29um69g/?tag=UltraWiderl) e uso um notebok acer) Estou gostando muito, principalmente pq estou iniciando meus estudos agora consigo fazer bastante coisas ao mesmo tempo, ler uma documentação enquanto vou testando as funções, assistir um video... E uma das melhores coisas q eu vi nele foi a opção de usar dois pc ao mesmo tempo nesses dias conectei um mac e um windows ao mesmo tempo kkkk

**Comment by kvnol** [comment]
Eu uso o [Teclado Microsoft Wired 600](https://www.kabum.com.br/produto/6881/teclado-multimidia-microsoft-wired-600-black-anb-00005/?tag=microsoft%20wired%20600) há pouco mais de um ano e nunca passei raiva com ele, apesar de ser básico.

**Comment by silvaMatheus** [comment]
Bom vou falar meu setup atual ( acho q não  posso dizer q estou na área ainda pois só estou estudando ainda) mas esta me servindo muito bem .
Os principais programas q eu uso atualmente para a parte de UI é o Photoshop e o XD, para a parte de programação to usando o SUBLIME  e o ATOM (estou vendo qual eu gosto mais) e navegadores Chrome e o Firefox  com esse note nunca chegou a travar ..

Monitor:  [Monitor Gamer LG 29 Full HD IPS LED 1ms UltraWide 21:9 HDMI](https://www.kabum.com.br/produto/90224/monitor-gamer-lg-29-full-hd-ips-led-1ms-ultrawide-21-9-hdmi-preto-29um69g/?tag=UltraWiderl)
Notebook: [Notebook Acer Aspire E5-573G-74Q5 Intel Core i7 8GB ](https://www.submarino.com.br/produto/125223582/notebook-acer-aspire-e5-573g-74q5-intel-core-i7-8gb-2gb-de-memoria-dedicada-1tb-led-15-6-windows-10-grafite?loja=03&opn=XMLGOOGLE&epar=bp_pl_00_go_g35213&WT.srch=1&gclid=Cj0KCQiAi7XQBRDnARIsANeLIeuKfnAa3M30ti3z-habewQUE5XPQD1mILzpcHOoyYujAV6aKErHznUaAh3XEALw_wcB)
Teclado: Uso o do note junto com um [suporte pro pulso](https://www.shoptime.com.br/produto/12635613/apoio-ergonomico-de-pulso-punho-c-base-p-teclado-digitador?opn=GOOGLEXML&loja=54651716000188&WT.srch=1&epar=bp_pl_00_go_pla_geral_3p&epar=bp_pl_00_go_pla_geral_3p&gclid=Cj0KCQiAi7XQBRDnARIsANeLIetLA-MSM2PjZ_LfzQLVMvE3ilzY8sbX2Z_2gOumFk1UXZjLxI5S_5UaAk9sEALw_wcB)
Mouse: Razer Deathadder
Fone: Razer kraken.

Obs: To pensando em pegar um macbook até final de 2018 mas a grana esta curta.

**Comment by ricardogouveia3** [comment]
Meu monitor extra é um [LG FullHD 24MN43](http://www.lg.com/br/monitores/lg-24MN43D). Acho que como ele não é novo, é difícil achar dele pra vender, mas tem outros equivalentes com preço parecido (aqui custou 890).
Como computador eu uso um MBP2016, que é bem caro mas vale totalmente a pena. Como o preço é limitante, se você quiser um apple pode tentar pegar um Macbook White (Fim de 2009 ou 2010) em bom estado, ou algum outro mais novo que esteja com preço similar (Da linha PRO, os 2011 não tem mais suporte)(Não compre nada com menos de 256GB de HD/SSD). Eu usei um White por mais de um ano antes do atual, e me servia muito bem (e ele quebrou por culpa minha usando fonte pirata).

**Comment by Vitor-Felipe** [comment]
@kvnol em relação ao monitor, creio que uma boa opção seria o **LG ultrawide 25"** [https://goo.gl/p1rwWT](url) o preço dele está muito próximo desse **LG LED 23"** e para desenvolver o tamanho dele é excelente! Além disso você pode comprar 2 **LG ultrawide 25"** com o valor de um **Monitor gamer** que o @silvaMatheus  indicou.
Porém as configurações do **Monitor gamer** são melhores que a do **LG ultrawide 25"**, mas mesmo assim ele é uma ótima opção.

**Comment by willycamargo** [comment]
Percebi que o tópico é exclusivamente voltado para computadores, porém ainda acho válido dar minha opinião.

Computadores desktops nunca servirão para mim. Desde 2011, quando comecei a trabalhar com TI, sempre tive a necessidade de trabalhar em lugares diferentes, portanto sempre foi indispensável ter um notebook.

## Notebook

Obviamente não tive condições de ter um macbook logo de cara, então meu primeiro notebook foi um **dell xps l502x**, que paguei em 12x no cartão de crédito 😄 

Usava ele com Ubuntu + maquina virtual com Windows XP otimizado para rodar softwares adobe. Depois de montado o setup, nunca tive sérios problemas.

Em 2016 surgiu a necessidade de trocar o computador, foi quando vendi esse dell e comprei meu primeiro **Macbook pro mid 15" 2015** (usado). Estou com o macbook até agora e sem dúvidas não tenho nenhum pingo de arrependimento.

Se você é uma pessoa que estuda e trabalha, o macbook será um investimento com ótimos retornos.

## Desktop

Pelas empresas que passei eu nem sempre usava meu próprio computador para trabalhar, portanto sempre havia um desktop disponível para eu poder trabalhar. Foram eles:

* Computador simples i5, 8gb de RAM, 500gb HD + Ubuntu + Monitor 22" Full HD Antireflexo (não lembro qual monitor exatamente) + Teclado e Mouse logitech.

* iMac 21" i5 mid 2010, 8gb RAM, 500gb HD (Monitor integrado) + Teclado Apple. Eu definitivamente não recomendo esse computador. A tela tem muito reflexo e sempre me incomodou.

* Mac mini i5, 12gb RAM, 500gb HD + Monitor AOC e2270swn (monitor simples mas suficiente)  + Teclado Apple.

## Minhas considerações finais

Eu acho o setup importante sim, ter um bom hardware, um bom monitor e um bom teclado e mouse é essencial para garantir conforto no dia a dia. Mas no geral eu acho que o sistema operacional pode influenciar muito mais do que tudo isso.

Sei da dificuldade em conseguir investir em um produto apple aqui no brasil, mas para trabalhar eu prefiro um mac mini usado do que um desktop novo.

> Aaah, mas no mac eu não consigo rodar meus jogos

Ok, você tem razão, mas eu estou falando sobre computador para trabalhar com desenvolvimento. Se jogar e trabalhar no mesmo computador é importante para você, então continue com seu desktop gamer.

**Comment by MathCunhaDev** [comment]
Eu por exemplo, sempre vi como questão fundamental a mobilidade, portanto, optei por um notebook ao invés de desktop. No momento estou com um Asus X555UB e um monitor auxiliar da Samsung. Não tenho nada a reclamar, roda tudo tranquilamente e ainda da pra jogar algumas coisas por causa da placa de vídeo. Chorei pro vendedor da FastShop e saiu por 2400, valeu muito a pena. 
15,6P Full HD
I5 6200U
Nvidia Geforce 940M - 2GB 
8GB Ram Dddr3 
HD 1 Tera

**Comment by romulosardinha** [comment]
Tenho um fraco por notebooks! Os meus outros desktops sempre me deram dor de kbeça, enquanto os notes nunca me deixaram na mão.

hoje trabalhando com um note Acer
i5 7200U 
8gb Ram
tela externa 22"
hd 1tb
plaquinha de vídeo qualquer intel 128
Mouse vertical recarregável.

Penei para encontrar uma boa configuração e o que deixar ligado ou não para melhor performance. 
chegando hoje em uma das melhores formas dele. usei varias dist linux, porem o win 10 o deixou na melhor forma. mesmo usando o chrome, vsc, psd e grover, não está perdendo em nada para os outros i7 que tenho quando trabalho com estes softwares abertos. Foi um ótimo investimento. (dica de últimos dias, não deixe o google drive sincronizando durante o trampo kkk 89% de uso de memória é de se assustar kkk)

**Comment by waldemirflj** [comment]
Tenho um MacPro Retina 13 2014 8GB, 256SSD e etc... Nunca me deu problemas e no começo dessa semana migrei para um Dell monstro.

Dell (com Ubuntu)
http://www.dell.com/br/p/inspiron-14-7460-laptop/pd?oc=cai7460w10he1852539brpw&model_id=inspiron-14-7460-laptop

**Comment by matheusml** [comment]
@kvnol 

Apesar do custo em geral ser bem menor em você montar sua própria máquina, uma das grandes vantagens dos produtos da Apple, é poder revender depois praticamente com o mesmo preço.

Eu nunca consegui revender um Samsung, um Dell, ou um HP. Mas revender um macbook é extremamente fácil.

Leve isso em consideração 👍

**Comment by thiagofrancisquete** [comment]
@willycamargo @matheusml @waldemirflj estou querendo comprar um macbook mas to meio perdido hehehe  poderiam me ajudar na escolha? o uso seria somente para estudo/trabalho, mas gostaria de ter um produto que dure uns bons anos rodando com um bom desempenho...

Um `Macbook Pro 256GB 2.3 GHz Intel Core I5` seria uma boa escolha? tenho essa duvida devido ele ser um i5 e não um i7, não sei se a diferença é muito grande... 

Um de `128gb` é muito pouco? 

E o `Macbook Apple 512GB 1.3 GHz`, seria uma boa escolha? Ele vem com um processador de `1,3 GHz Intel Core m3 dual core de 1,3 GHz`, e sinceramente não conheço ele, mas me parece inferior a familia de processadores `i3/i5/i7`

Vocês compraram o de voces ou conhecem alguem que comprou no `mercado livre`? La tem opções do `mackbook pro i7 512gb` por 9k, bem mais em conta que os 15k da loja da apple ou os 12k dos de 256gb

**Comment by willycamargo** [comment]
@thiagofrancisquete

Eu tenho um 15" e hoje, particularmente, compraria um de 13". 15 é bastante pesado e pra mim não tem tanta vantagem assim, pois acaba perdendo um pouco a portabilidade (que é a maior vantagem de um notebook, né?).

Eu tenho 256gb SSD e não me falta em nada, mas vale ressaltar que isso vai de pessoa pra pessoa. No pior das hipóteses (no caso de 128gb) você pode usar o espaço interno apenas para softwares e pouco armazenamento e, o maior, você deixa na Nuvem ou em um HD externo.

Sobre processadores, tanto o `2.3 GHz Intel Core I5` quanto o `2.2 GHz Intel Core i7` vão dar conta do recado para trabalhar com desenvolvimento (mas recomendo pelo menos 8gb de RAM), já o `Intel Core m3 dual core de 1,3 GHz` eu não tenho certeza se vale a pena.

Sobre comprar no **mercado livre**: não vejo problema nenhum. Já fiz várias compras lá e só não comprei meu macbook lá porque apareceu uma oportunidade melhor aqui na minha cidade (Encontrei no OLX por 6.500 com possibilidade de parcelar sem juros)

É isso ai, abrs!

**Comment by kelciocajueiro** [comment]
Meio off, mas quem for comprar Macbook no Mercado Livre não compre com a loja chamada Sultech Informática. Comprei com eles e tive dor de cabeça.

**Comment by felipefialho** [comment]
Acho que não tenho um desktop desde 2009, hehe, to totalmente por fora.

Acredito que o @grippado e o @mtmr0x podem ajudar nisso.

@willycamargo Muito boa sua resposta, seria legal uma issue sobre notebooks heim?

**Comment by grippado** [comment]
##### Opa, valeu a citação @LFeh, atualmente eu tenho um desktop, que é um híbrido entre trabalho e jogos, portanto montei uma maquina meio forte (pra não dizer estúpida), vou mandar minhas specs, e colocar dentro do contexto que o @kvnol criou, como indicações para a tebela dele!

**Monitor**: 288P6LJEB/57 (é um Philips 4k/UHD com PiP, e 1ms GtG e 5ms default) eu recomendo esse monitor como avançado pra tabela, por ser simplesmente animal ter um monitor 4k pra usar no dia a dia, se ainda não teve a oportunidade, tente experimentar pra entender o que é, e prepare-se para readaptar o tamanho das fontes do seu editor e do seu terminal =)

**Teclado**: Coolermaster Masterkeys Pro S (é um teclado mecânico voltado para jogos, mas que é extremamente confortável para digitação e coding). Sim, eu recomendo fortemente um teclado mecânico pra uso diário, o conforto e a precisão de teclas com acionamento individual, é outro ponto que recomendo experimentar pra conhecer (preferencia por switchs com resposta tactil, "blue" e "brown") 

**Mouse**:  Acho que da pra dissertar longamente sobre tipo de pegada de mouse (sim, isso importa muito pra ergonomia de uso, seja pra jogos, ou uso de rotina, portanto, acho que mouse, vc pode pesquisar melhor qual o tipo de ergonomia vc necessita, então vão indicar nenhum em específico, mas vou colocar aqui os que eu uso:
- pra jogar, uso um Razer Krait 4G (porte médio, pegada claw)
- pra dia a dia, uso um Microsoft Arc Touch (eu amo esse mouse, de verdade, tipo muito)

**Processador**: Aqui, o que vai importar é o bolso, principalmente. Eu tenho um AMD FX E-8350e (8-core, 4Ghz usual, nada de turbi boost, etc), e ele é um canhão de desempenho, quando comprei, ele fazia frente ao i7-6700k, só que custando 60% do valor do mesmo, não me decepcionei, e recomendo a linha Ryzen da AMD ou a propria linha i5/i7 da intel (se vc não tem limites pra gastar, compra um i9 e me chama pra experimentar)

**Fonte**: Esse é somente um detalhe técnico, que vc vai comprar de acordo com todas as outras partes do computador que vc montar (principalmente da placa de vídeo, item que percebi ser faltante nessa tabela, mas que vou adicionar aqui). Se vc leu até a parte que falei da minha GPU, leia aqui novamente: eu tenho uma CX750 da corsair, que é uma fonte parruda, que suporta a config que eu uso, mas reitero, isso vai ser muito da config que vc montar, para as placas mais leves (R3 230 e GT1030) que citei, uma fonte de 350 ou 400w vão ser suficientes, pois elas nem precisam de alimentação dedicada, igual a uma placa mais ~estúpida~.

**GPU**: Sim, GPU pra programar, pq vc acha que se aplicar tudo que o Sergio Caelum mostrou de tirar o processamento da Main Thread do browser, e jogar pra GPU, aquela sua HD250 do seu Core2duo do macbook 2007 que te indicaram ser topissimo, vai aguentar a quantidade de merda que todos nos fazemos dia a dia no desenvolvimento? ~quem não testa e viu que ficou uma bosta até fazer melhor não é mesmo?~.. sobre GPUs, não precisa comprar uma GTX 1080 de 2 mil reais pra codar, mas se vc comprar qualquer uma que seja, dedicada, um Radeon R5 230 ou uma GeForce GT1030 (recomendo essa duas como placas básicas/médias na tabela), seus problemas com possíveis gargalos de processamento gráfico vão ser sanados, e já que vc ta montando um desktop, com certeza uma placa de video dedicada, da mais básica que seja, é sim, importante. Eu tenho uma RX 470 de 6Gb, pois como falei no inicio, meu uso é híbrido de jogos/trampo, portante, foi algo que investi um pouco mais.

**Cooler**: Mesmo caso da fonte, vai ser muito da sua necessidade, e normalmente, quando se compra um processador em modo BOX, ele já vem com um cooler que é o próprio e recomendado para um uso inclusive de jogos, do próprio processador, portante, só se vc pensar em comprar uma maquina que também será seu console, vc terá de se preocupar com esse item (eu uso o cooler original do meu processador, com outros 4 exaustores somente para tirar o ar quente de dentro da CPU, isso é realmente algo completamente pessoal)

**Placa mãe**: ~vou me abster de falar de placa mãe pq ficaria longo demais~ (sim, muito mais)

**HDD/SSD**: um outro ponto importantíssimo que não foi colocado na tabela, mas influencia absurdamente, e eu só tenho um comentário:
### COLOQUEM UM SSD NA MAQUINA QUE ESTIVEREM MONTANDO, SEJA DE 60/120GB O QUE FOR SOMENTE PRA RODAR O SISTEMA OPERACIONAL, MAS COLOQUEM, QUE SIM, FAZ MUITA DIFERENÇA!
(edit adicionado com sucesso)


bom, é isso, espero que seja útil de alguma forma para a thread, e boa iniciativa @kvnol, mandando muito como sempre! BANG

**Comment by matheusmcp** [comment]
Estou começando a estudar front-end agora, e estou entre:

Concertar (ou tentar) um **Notebook** LG-P42 de 14 polegadas, com um i3, 4GB de RAM (8GB posteriormente).

**Desktop:** Com um i5 3750 (que compraria com o dinheiro do reparo do notebook), 4GB de RAM (8GB posteriormente), monitor AOC e970swnl de 18 polegadas.

A resposta seria óbvia se não colocasse na balança a mobilidade de um notebook, e o fato do desktop que está aqui ser o único da casa, que tem quatro pessoas que usam, por isso estou com essa dúvida. 

OBS: As memórias rodam em 1333mhz e sem dissipador, isso pode afetar o desempenho? Minha placa mãe e o notebook só aceitam até 16GB de RAM funcionando a 1600mhz.

**Comment by caioincau** [comment]
Minha dica é, hoje em dia, desenvolver com menos de 8GB de RAM é sofrido, provavelmente tu vai ficar fazendo watch de arquivos e um monte de outras coisas.

Hoje em dia meu processador fica o tempo todo em 30~50%, com tudo aberto(i5 de notebook), mas a RAM tá sempre com ~6GB usados.

**Comment by matheusmcp** [comment]
@caioincau Obrigado pela resposta. Pelo visto é melhor pegar o i5 pro meu desktop mesmo, e mês que vem comprar 2 pentes de 4GB de RAM.

**Comment by AlexAugustoM** [comment]
Eu estou iniciando também, os programas que mais uso pra codar é o Pycharm, Intellij IDEA, e de vez em quando o Android Studio, mas estou testado também o Visual Studio Code, navegadores são o Chrome mesmo, e só trava quando eu ligo o emulador no Android Studio.
Notebook Acer Aspire E1-471-6413 que ganhei em 2013, tem um I3-2328m com 8GB de RAM (sim, só trava quando ligo emulador no Android Studio kkk) e tirei o HD e coloquei um SSD de 480GB, monitor eu uso só a tela dele e um mouse Fortrek Raptor

**Comment by DyogoViana** [comment]
man... eu não me arrependo de ter comprado o Thinkpad T480.
só mudo quando quebrar ou para comprar um Thinkpad X1.

Achei melhor que um Mac. 
Ele entrega uma ótima experiência no uso, além de ser resistente e tantas outras coisas. 

Ele perde na tela, mas mesmo assim ele resolve bem a parte de design e para desenvolvimento ele n peca em nada.
